ibmcloud/README.md

```diff
@@ -160,75 +160,68 @@ You need to build a pod VM image for peer pod VMs. A pod VM image contains the f
+
+Create the `terraform.tfvars` in [the template directory](./terraform/podvm-build). `terraform.tfvars` should look like this.
 
 ```
-ibmcloud login -a $IBMCLOUD_API_ENDPOINT -r $IBMCLOUD_VPC_REGION -apikey $IBMCLOUD_API_KEY
-COS_INSTANCE_GUID=$(ibmcloud resource service-instance --output json "$IBMCLOUD_COS_SERVICE_INSTANCE" | jq -r '.[].guid')
-ibmcloud iam authorization-policy-create is cloud-object-storage Reader --source-resource-type image --target-service-instance-id $COS_INSTANCE_GUID
+ibmcloud_api_key = "<your API key>"
+ibmcloud_user_id = "<IBM Cloud User ID>"
+cluster_name = "<cluster name>"
+cos_service_instance_name = "<COS Service Instance Name>"
+cos_bucket_name = "<COS Bucket Name>"
 ```
 
-Then, you can execute the image upload script by using `make`.
+If you used the Terraform templates in [common](./terraform/common) and [cluster](./terraform/cluster) to create the VPC and VSIs, you should set `ibmcloud_api_key` and `cluster_name` to the same values as those you entered in `terraform.tfvars` for those templates.
+
+`ibmcloud_user_id` should be set to the IBM Cloud user ID who owns the API key `ibmcloud_api_key`. You can look up the user ID using.
 
 ```
-CLOUD_PROVIDER=ibmcloud make push
+ibmcloud account users
 ```
 
-After successfully uploading an image, you can verify the image by creating a virtual server instance using it.
+If `ibmcloud account users` displays multiple user IDs, choose the user ID whose state is `ACTIVE`.
 
-**Note:** The `Operator` and `Console Admin` roles must be [assigned](https://cloud.ibm.com/docs/vpc?topic=vpc-vsi_is_connecting_console&interface=ui) to the user.
+The `cos_service_instance_name` and `cos_bucket_name` tfvars are optional and default to values based on your `cluster_name` tfvar if not set, as per the example in the following table.
 
+| Resource | Value |
+|----------|-------|
+| `cluster_name` tfvar | peer-pod-cluster |
+| COS Service Instance name | peer-pod-cluster-cos-service-instance | 
+| COS Bucket name | peer-pod-cluster-cos-bucket |
 
-The following command will create a new server, and delete it. The VPC and subnet name are available in the terraform configuration mentioned above. You need to change the zone name if you have changed the region.
+For uploding the pod VM image using Terraform, the COS Bucket must be a regional bucket in the same region (default `jp-tok`) as the VPC and VSIs.
 
-```
-export IBMCLOUD_VPC_NAME=tok-vpc
-export IBMCLOUD_VPC_SUBNET_NAME=tok-primary-subnet
-export IBMCLOUD_VPC_ZONE=jp-tok-2
+**Note:** The `Operator` and `Console Admin` roles must be [assigned](https://cloud.ibm.com/docs/vpc?topic=vpc-vsi_is_connecting_console&interface=ui) to the user. The Terraform template will create the `Console Admin` role for the user `ibmcloud_user_id` is set to in the template `terraform.tfvars`.
```

Comment:
If I already assign the `Console Admin` role by hand.
It will report:
```
╷
│ Error: The policy wasn't created because an access policy with identical attributes already exists. Please update the roles in the existing policy (857ef95b-35fa-45e4-8673-8ca069252be6), or update the one you're trying to assign to include a different attribute assignment.
│
│   with ibm_iam_user_policy.is_console_administrator_policy,
│   on main.tf line 56, in resource "ibm_iam_user_policy" "is_console_administrator_policy":
│   56: resource "ibm_iam_user_policy" "is_console_administrator_policy" {
│
╵
```
Can we do some check here? If the role already there then we skip the create role task, otherwise we create the `Console Admin` role

Reply:
I'm working on an elegant way to only create the ibm_iam_user_policy.is_console_administrator_policy resource if an iam user policy with the right policy does not already exist.

Reply:
Made it so that the podvm Terraform template won't try and create an Console Administrator user policy for all VPC Infrastructure resources if such as policy already exists.